File: mobile/apps/photos/plugins/ente_photos_platform/android/src/main/kotlin/io/ente/photos/platform/PhotosPlatformPlugin.kt
package io.ente.photos.platform

import io.ente.photos.platform.flutter.CountryNamesChannelAdapter
import io.ente.photos.platform.flutter.DeviceHealthChannelAdapter
import io.ente.photos.platform.flutter.DeviceTrashChannelAdapter
import io.ente.photos.platform.flutter.ProcessLockChannelAdapter
import io.flutter.embedding.engine.plugins.FlutterPlugin

class PhotosPlatformPlugin : FlutterPlugin {
    private val countryNamesAdapter = CountryNamesChannelAdapter()
    private val deviceHealthAdapter = DeviceHealthChannelAdapter()
    private val deviceTrashAdapter = DeviceTrashChannelAdapter()
    private val processLockAdapter = ProcessLockChannelAdapter()

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        countryNamesAdapter.attach(binding)
        deviceHealthAdapter.attach(binding)
        deviceTrashAdapter.attach(binding)
        processLockAdapter.attach(binding)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        countryNamesAdapter.detach()
        deviceHealthAdapter.detach()
        deviceTrashAdapter.detach()
        processLockAdapter.detach()
    }
}

File: mobile/apps/photos/plugins/ente_photos_platform/ios/Flutter/PhotosPlatformPlugin.swift
@preconcurrency import Flutter
import Foundation

@MainActor
public final class PhotosPlatformPlugin: NSObject, @preconcurrency FlutterPlugin {
    private let countryNamesAdapter: CountryNamesChannelAdapter
    private let deviceHealthAdapter: DeviceHealthChannelAdapter
    private let processLockAdapter: ProcessLockChannelAdapter

    private init(registrar: FlutterPluginRegistrar) {
        countryNamesAdapter = CountryNamesChannelAdapter(registrar: registrar)
        deviceHealthAdapter = DeviceHealthChannelAdapter(registrar: registrar)
        processLockAdapter = ProcessLockChannelAdapter(registrar: registrar)
        super.init()
    }

    public static func register(with registrar: FlutterPluginRegistrar) {
        registrar.publish(PhotosPlatformPlugin(registrar: registrar))
    }

    public func detachFromEngine(for registrar: FlutterPluginRegistrar) {
        countryNamesAdapter.detach()
        deviceHealthAdapter.detach()
        processLockAdapter.detach()
    }
}
